sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -755,6 +755,9 @@ case class MapObjects private(
   }
 
   private lazy val mapElements: Seq[_] => Any = customCollectionCls match {
+    case Some(cls) if classOf[WrappedArray[_]].isAssignableFrom(cls) =>
```

Comment:
does it really work with any sub-class of `WrappedArray`?

Reply:
Yes. We use `WrappedArray.make()` below to generate the sub-calss of  `WrappedArray` and `make()` supports all the sub-calsses.